### PR TITLE
[Doc] Rename Architecture page to Key Concepts

### PR DESCRIPTION
**Pull request by fzaninotto**
## Problem

Newcomers often feel a bit lost after completing the tutorial, as they don't really know how react-admin works, or to structure their app.

## Solution

- Expose the key concepts of react-admin architecture sooner in the documentation
- Include best practices

### CODE REVIEW

File: docs/Architecture.md [hunk 1]
```diff
@@ -90,6 +222,110 @@ const App = () => (
...
+
+React-admin exposes [dozens of hooks](./Reference.md#hooks) to help you build your own components. You can even build an entire react-admin application without ever using the material-ui components, and use another UI kit instead.
+
+## Context: Pull, Don't Push
+
+Communicating between components is a common problem in React applications, especially in large ones, when you have to pass props down several levels. React-admin solves this problem by using a pull model: components expose props to their descendants via a context, and descendants can use them via a custom hook.
+
+Whenever a react-admin component fetches data or defines a callback, the component creates a context and puts the data and callback in it.
+
+For instance, the `<Admin>` component creates an `I18NProviderContext`, which exposes the `translate` function. All components in the application can use the `useTranslate` hok, which reads the `I18NProviderContext`, to translate their labels and messages.
```
Comment (adguernier) [inline docs/Architecture.md:281]:
```suggestion
For instance, the `<Admin>` component creates an `I18NProviderContext`, which exposes the `translate` function. All components in the application can use the `useTranslate` hook, which reads the `I18NProviderContext`, to translate their labels and messages. 
```

File: docs/Architecture.md [hunk 2]
```diff
@@ -90,6 +222,110 @@ const App = () => (
...
+            attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
+            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+            />
+            <Marker position={record[source]} />
+        </MapContainer>
+    );
+};
+
+const StoreShowPage = () => (
+    <Show>
```
Comment (adguernier) [inline docs/Architecture.md:316]:
```suggestion
    <Show> // create a RecordContext
```

File: docs/Architecture.md [hunk 3]
```diff
@@ -90,6 +222,110 @@ const App = () => (
...
+```
+
+In a similar way, the `<Show>` component fetches a record and exposes it via a `RecordContext`. Inside the `<Show>` component, you can use the `useRecordContext` hook to access the record - for instance to display a map of the record location.
+
+```jsx
+import { useRecordContext } from 'react-admin';
+import { MapContainer, TileLayer, Marker } from 'react-leaflet';
+
+const LocationField = ({ source }) => {
+    const record = useRecordContext(props);
```
Comment (adguernier) [inline docs/Architecture.md:301]:
```suggestion
    const record = useRecordContext(props); // use the RecordContext created by <Show> 
```

File: docs/Architecture.md [hunk 4]
```diff
@@ -90,6 +222,110 @@ const App = () => (
...
+            <LocationField source="location" />
+        </SimpleShowLayout>
+    </Show>
+)
+```
+
+This simple approach removes the need for a dependency injection system. 
+
+So when you write a component that need to access data or callbacks defined higher in the render tree, you can always find a context to get it. 
+
```
Comment (adguernier) [inline docs/Architecture.md:328]:
```suggestion
Contexts are one of the key concepts in React Admin. If you are not familiar with this, do not hesitate to read the [official documentation](https://react.dev/learn/passing-data-deeply-with-context).
```

File: docs/Architecture.md [hunk 5]
```diff
@@ -88,96 +212,231 @@ const App = () => (
...
 
-React-admin contexts aren't exposed directly. Instead, react-admin exposes hooks to access the context content. In addition, the framework also packages bits of reusable logic as hooks, to facilitate the customization of the UI of existing components without having to rewrite everything. Finally, hooks hide the implementation details of the framework, so that you can focus on the business logic.
+So when you write a component that needs to access data or callbacks defined higher in the render tree, you can always find a context to get it. 
 
-So hooks are the primary way to read and change a react-admin application state. We use them in almost every react-admin component, and it's perfectly normal to use react-admin hooks in your own components.  
+Contexts are fundamental concepts in React Admin. If you are not familiar with them, don't hesitate to read the [React documentation on Context](https://react.dev/learn/passing-data-deeply-with-context). Understanding contexts will greatly enhance your understanding of how react-admin leverages them to create a powerful and flexible framework.
```
Comment (adguernier) [inline docs/Architecture.md:321]: 👍

File: docs/Architecture.md [hunk 6]
```diff
@@ -88,96 +212,231 @@ const App = () => (
...
 
-For instance, the `useRefresh` hook packages the logic to refetch the data currently displayed on the screen. Developers don't need to know how it works, just how to use it:
+## Batteries Included But Removable
 
-```jsx
-import { useRefresh } from 'react-admin';
+React-admin allows you to build sophisticated web applications using only its built-in components, assuming that its design choices align with your needs. However, if you find that a component's existing capabilities don't meet your specific requirements, you're free to replace it with a custom component.
```
Comment (adguernier) [inline docs/Architecture.md:325]:
```suggestion
React-admin allows you to build sophisticated web applications using only its built-in components, assuming that its design choices align with your needs. However, if you find that a component's existing capabilities don't meet your specific requirements, feel free to replace it with a custom component.
```

**Review by adguernier — APPROVED** [review]

File: docs/Architecture.md (hunk 6 above)
Comment (fzaninotto) [reply docs/Architecture.md:325]: Not sure this is better.

**Pull request merged**